HTTP_HEADER_FORMAT.md

```diff
@@ -61,26 +60,97 @@ caller rather than strict rules to follow for 3 reasons:
 3.  Different load between caller service and callee service might force callee
     to down sample.
 
-The behavior of other bits is currently undefined.
+### Prefix '.' The Hierarchical-id
 
-#### Examples of HTTP headers
+The '.' character indicates that a variable number of characters follow and represent
+a hierarchical (path based) id.   It has the form of a list of hexadecimal numbers 
+(lower case) separated by '.' characters.   The list is optionally terminated by 
+a '!' character.   The id ends if either the string ends or a character that 
+is not a hexadecimal character or a '.' or '!' is encountered.   
 
-*Valid sampled Trace-Context:*
+Examples
 
 ```
-Value = 00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01
-base16(<Version>) = 00
-base16(<TraceId>) = 4bf92f3577b34da6a3ce929d0e0e4736
-base16(<SpanId>) = 00f067aa0ba902b7
-base16(<TraceOptions>) = 01  // sampled
+	.1
+	.3.b.34.ca25.56
+	.334.ba34.93a4.1.1!
 ```
+The hierarchical ID is meant to represent series of causal requests within the scope
+of a given Trace-id.   Take has a whole, like the Span-ID, combined with the Trace-id
+it uniquely identifies the request.    However in addition the components also represent
+other requests with that same trace that are causality connects.  For example the Trace-Context
+value of 
+```
+*4bf92f3577b34da6a3ce929d0e0e4736.3.b.34
+```
+Represents a request with ID
+```
+*4bf92f3577b34da6a3ce929d0e0e4736.3.b.34
+```
+that was caused by a request with ID
+```
+*4bf92f3577b34da6a3ce929d0e0e4736.3.b
```

Comment:
requestID == spanID, is it not? Instead it looks like the hierarchical parts are appended to the traceID, which should be stable across all hops

Reply:
I am not sure what your question is.  However in its simplest use you would only have a Trace-ID value and a hierarchical-ID value in in a system that employs hierarchical IDs.

What may be causing confusion here is that we are not trying to describe a single system. but a framework by which DIFFERENT systems can coexists to the extent possible.   Thus some systems might use Trace-id Span-id pairs to represent a request ID and other systems will use a Trace-id - hierarchical-id to do so.     

It is still an open question how much interoperability is feasible in a system compost of both types of logging conventions, but a assumption of this specification is that making it possible to share the concept of TraceId, is huge step forward and is likely to be required for any interesting interoperability.  

> Instead it looks like the hierarchical parts are appended to the traceID, which should be stable across all hops

Yes the hierarchical-id is that takes the common trace-id and makes a particular request unique (within that trace)

Reply:
My understanding of Microsoft spec that mentioned hierarchical-id was that it still had a stable "trace id" (that was suggested to be passed in the "baggage" header), which was different from unique ID assigned to each request (span-id). Even if it's not correct and the hierarchical id contains the stable part, why can't that stable part be separated into a "trace-id" field understood by this spec?

Reply:
> My understanding of Microsoft spec that mentioned hierarchical-id was that it still had a stable "trace id" 

No, in this spec, hierarchical-id does not include Trace-id, but it expected to be used in conjunction with it.   Thus the Trace-Context value 

*4bf92f3577b34da6a3ce929d0e0e4736.3.b 

Has a trace ID in it 4bf92f3577b34da6a3ce929d0e0e4736 (which is table for everything caused by this), and  3.b which is the hierarchical-id (which indicates a particular request within the scope of the Trace-id.   

Now of course the full request ID in Microsoft's hierarchical system would include the Trace-id, but in this spec a least 'hierarchical-id' refers only to the part that excluded the Trace-id.

Reply:
great, so why not `*4bf92f3577b34da6a3ce929d0e0e4736-3.b`, thus clearly separating the stable portion into the trace-id field? This way the receiving instrumentation doesn't need to even understand hierarchical IDs as long as it can record `3.b` as correlation and is able to reuse `4bf92f3577b34da6a3ce929d0e0e4736` as the trace id.

Reply:
> great, so why not *4bf92f3577b34da6a3ce929d0e0e4736-3.b 

Because the - prefix is already used to represent a 16 character span id.   (the whole idea behind the prefixes, is to allow you to 'mix and match' parts of the value in this Header field.